app/src/main/java/com/example/myapplication/ui/home/HomePresenter.kt

```diff
@@ -1,6 +1,12 @@
-package com.example.myapplication.ui
+package com.example.myapplication.ui.home
 
-class Presenter( private val mainFragment: BaseContract.View) : BaseContract.Presenter {
+class HomePresenter(private val mainFragment: HomeContract.View) :
```

Comment:
Prezenter nie zawsze będzie otrzymywał fragment, czasem będzie to dialog lub zwykły View :) Warto zatem uogólnić ten argument do `view: HomeContract.View`